engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/unboxing/SuspendedFieldGetterNode.java

```diff
@@ -35,32 +39,28 @@ static UnboxingAtom.FieldGetterNode build(UnboxingAtom.FieldGetterNode get, Unbo
     return new SuspendedFieldGetterNode(get, set);
   }
 
+  private static boolean shallBeExtracted(Function fn) {
+    return fn.isThunk() || ReadArgumentCheckNode.isWrappedThunk(fn);
+  }
+
   @Override
   public Object execute(Atom atom) {
     java.lang.Object value = get.execute(atom);
-    if (value instanceof Function fn && fn.isThunk()) {
+    if (value instanceof Function fn && shallBeExtracted(fn)) {
       try {
         org.enso.interpreter.runtime.EnsoContext ctx = EnsoContext.get(this);
         java.lang.Object newValue = invoke.execute(fn, null, State.create(ctx), new Object[0]);
         set.execute(atom, newValue);
         return newValue;
       } catch (AbstractTruffleException ex) {
-        var rethrow = new SuspendedException(ex);
+        var rethrow = DataflowError.withTrace(ex, ex);
         set.execute(atom, rethrow);
```

Comment:
`set.execute` was failing with `SuspendedException` because the _lazy field are runtime type checked_ and `SuspendedException` didn't have the right type ;-)

Changing into `DataflowError` as that is `isAllFitValue`.